keras/callbacks.py

```diff
@@ -227,6 +228,33 @@ def on_epoch_end(self, epoch, logs=None):
                     logs[k] = self.totals[k] / self.seen
 
 
+class TerminateOnNaN(Callback):
+    '''Callback that terminates training when a nan loss is encountered.
+
+    # Arguments
+        patience: number of mini-batches with invalid loss
+            after which training will be stopped.
+    '''
+    def __init__(self, patience=0):
+        super(TerminateOnNaN, self).__init__()
+        self.patience = patience
+        self.invalid = 0
+
+    def on_batch_end(self, batch, logs={}):
+        # terminate training immediately if nan loss encountered in a batch
+        current = logs.get('loss')
+        if np.isnan(current) or np.isinf(current):
+            self.invalid += 1
+        else:
+            self.invalid = 0
+
+        try:
+            assert self.invalid <= self.patience
```

Comment:
`try` with an `assert` is not the way to test inequality between two integers...